packages/studio/src/components/AddPageButton/StreamScopeCollector.tsx

```diff
@@ -1,58 +1,185 @@
-import { useCallback, useContext } from "react";
-import FormModal, { FormData } from "../common/FormModal";
+import { useCallback, useContext, useEffect, useMemo } from "react";
+import DialogModal from "../common/DialogModal";
 import { FlowStepModalProps } from "./FlowStep";
 import AddPageContext from "./AddPageContext";
-import StreamScopeParser, {
-  StreamScopeForm,
-} from "../../utils/StreamScopeParser";
-
-export const streamScopeFormData: FormData<StreamScopeForm> = {
-  entityIds: {
-    description: "Entity IDs",
-    optional: true,
-    tooltip: "In the Yext platform, navigate to Content > Entities",
-  },
-  entityTypes: {
-    description: "Entity Type IDs",
-    optional: true,
-    tooltip:
-      "In the Yext platform, navigate to Content > Configuration > Entity Types",
-  },
-  savedFilterIds: {
-    description: "Saved Filter IDs",
-    optional: true,
-    tooltip:
-      "In the Yext platform, navigate to Content > Configuration > Saved Filters",
-  },
-};
+import { Tooltip } from "react-tooltip";
+import useStudioStore from "../../store/useStudioStore";
+import { StreamScope } from "@yext/studio-plugin";
+import PillPickerInput from "../PillPicker/PillPickerInput";
+import { streamScopeFormData } from "../PageSettingsButton/EntityPageModal";
 
 export default function StreamScopeCollector({
   isOpen,
   handleClose,
   handleConfirm,
 }: FlowStepModalProps) {
-  const { actions } = useContext(AddPageContext);
-
-  const onConfirm = useCallback(
-    async (data: StreamScopeForm) => {
-      const streamScope = StreamScopeParser.parseStreamScope(data);
-      actions.setStreamScope(streamScope);
-      await handleConfirm();
-      return true;
+  const { state, actions } = useContext(AddPageContext);
+  const { streamScope } = state;
+  const { setStreamScope } = actions;
+  const [savedFilters, entitiesRecord] = useStudioStore((store) => [
+    store.accountContent.savedFilters,
+    store.accountContent.entitiesRecord,
+  ]);
+
+  useEffect(() => {
+    if (streamScope === undefined) {
+      setStreamScope({});
+    }
+  }, [setStreamScope, streamScope]);
+
+  const updateScopeItems = useCallback(
+    (field: string) => (selectedItems: string[] | undefined) => {
+      const updatedStreamScope = {
+        ...streamScope,
+        [field]: selectedItems,
```

Comment:
I was under the impression that Andrew didn't want a *non*-technical Studio user to be able to specify a "combined" filter for the scope. The disabling behavior in the UI that you mentioned should be sufficient to prevent that use case, particularly for the Add Page flow where the stream scope would never have any pre-existing filters defined.

For editing existing pages' template configs, if we wanted to stop technical users, we'd have to add a check while parsing their template config and have it give an error if filters of multiple types are specified. Technically, this particular function's logic wouldn't stop you from defining a "combined" filter. But, because I wasn't sure how we'd want to handle the technical user's case for existing pages, I thought it would be more graceful to leave in the merging of the existing stream scope.

But, since updating the Page Settings modal will be another item, I can also update this logic to strictly enforce the XOR for now if you prefer

Reply:
Whoops, I meant non-technical user! If you don't mind, let's enforce the XOR here as well. I think it's always good to program defensively if we can. Someone may inadvertently break the disabling logic below, but we'd still be safe if we enforce it here as well. If a technical person has a more complex, "combined" Scope filter, I think we shouldn't allow the non-technical user to edit it the Scope in any way. Whenever a non-technical person is able to make an edit, we should enforce they're only adding one type of filter.